server/repository/src/db_diesel/temperature_chart/temperature_chart_row.rs

```diff
@@ -0,0 +1,224 @@
+    }
+    fn all_columns() -> Self::AllColumns {
+        (
+            FromDatetime,
+            ToDatetime,
+            AverageTemperature,
+            TemperatureLogId,
+            SensorId,
+        )
+    }
+}
+impl AppearsInFromClause<TemperatureChart> for TemperatureChart {
+    type Count = Once;
+}
+impl AppearsInFromClause<TemperatureChart> for () {
+    type Count = Never;
+}
+// pub type BoxedQuery<'a, DB, ST = SqlType> = BoxedSelectStatement<'a, ST, TemperatureChart, DB>;
+
+impl AsQuery for TemperatureChart {
+    type SqlType = SqlType;
+    type Query = SelectStatement<Self>;
+    fn as_query(self) -> Self::Query {
+        SelectStatement::simple(self)
+    }
+}
+
+// This is just boilerplate taken from expanding table! macro
+macro_rules! temperature_chart_column {
+    ($column_name:ident, $column_sql: expr, $column_type:ty) => {
+        pub struct $column_name;
+        impl<DB> QueryFragment<DB> for $column_name
+        where
+            DB: Backend,
+            <TemperatureChart as QuerySource>::FromClause: QueryFragment<DB>,
+        {
+            fn walk_ast(&self, mut out: AstPass<DB>) -> QueryResult<()> {
+                out.push_sql($column_sql);
+                Ok(())
+            }
+        }
+        impl Expression for $column_name {
+            type SqlType = $column_type;
+        }
+        impl SelectableExpression<TemperatureChart> for $column_name {}
+        impl NonAggregate for $column_name {}
+        impl<QS> AppearsOnTable<QS> for $column_name where
+            QS: AppearsInFromClause<TemperatureChart, Count = Once>
+        {
+        }
+        impl Column for $column_name {
+            type Table = TemperatureChart;
+            const NAME: &'static str = $column_sql;
+        }
+
+        impl<From> SelectableExpression<SelectStatement<From>> for $column_name where
+            $column_name: SelectableExpression<From> + AppearsOnTable<SelectStatement<From>>
+        {
+        }
+
+        impl<T> EqAll<T> for $column_name
+        where
+            T: AsExpression<$column_type>,
+            diesel::dsl::Eq<$column_name, T>: Expression<SqlType = Bool>,
+        {
+            type Output = diesel::dsl::Eq<Self, T>;
+            fn eq_all(self, rhs: T) -> Self::Output {
+                diesel::expression::operators::Eq::new(self, rhs.as_expression())
+            }
+        }
+    };
+}
+
+temperature_chart_column!(FromDatetime, "time_series.from_datetime", Timestamp);
+temperature_chart_column!(ToDatetime, "time_series.to_datetime", Timestamp);
+temperature_chart_column!(
+    AverageTemperature,
+    "AVG(temperature_log.temperature) as average_temperature",
+    Double
+);
+temperature_chart_column!(TemperatureLogId, "temperature_log.id", Text);
+temperature_chart_column!(SensorId, "temperature_log.sensor_id", Text);
+
+// Needed to filter chart data by temperature log ids
+
+// Combined tests are done in temperature_chart repo
+#[cfg(not(feature = "postgres"))]
```

Comment:
a postgres version would be good too

Reply:
Great point: daaa597c384d90a38f792f0f944afd3f9ed03e85